src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -102,23 +100,35 @@ private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
                 {
                     context.ReportDiagnostic(diagnostic);
                 }
+            }
+        }
 
-                // Due to https://github.com/dotnet/roslyn/issues/41640, enabling this analyzer (IDE0005) on build requires users
-                // to enable generation of XML documentation comments. We detect if generation of XML documentation comments
-                // is disabled for this tree and IDE0005 diagnostics are being reported with effective severity "Warning" or "Error".
-                // If so, we report a special diagnostic that recommends the users to set "GenerateDocumentationFile" to "true"
-                // in their project file to enable IDE0005 on build.
-                if (tree.Options.DocumentationMode == DocumentationMode.Parse)
-                {
-                    var effectiveSeverity = descriptor.GetEffectiveSeverity(context.SemanticModel.Compilation.Options, tree, context.Options);
-                    if (effectiveSeverity is ReportDiagnostic.Warn or ReportDiagnostic.Error)
-                    {
-                        context.ReportDiagnostic(Diagnostic.Create(s_enableGenerateDocumentationFileIdDescriptor, diagnostics.First().Location));
-                    }
-                }
+        private void AnalyzeCompilation(CompilationAnalysisContext context)
+        {
+            // Due to https://github.com/dotnet/roslyn/issues/41640, enabling this analyzer (IDE0005) on build requires users
+            // to enable generation of XML documentation comments. We detect if generation of XML documentation comments
+            // is disabled for this tree and IDE0005 diagnostics are being reported with effective severity "Warning" or "Error".
+            // If so, we report a special diagnostic that recommends the users to set "GenerateDocumentationFile" to "true"
+            // in their project file to enable IDE0005 on build.
+
+            var compilation = context.Compilation;
+            var tree = compilation.SyntaxTrees.FirstOrDefault();
+            if (tree is null || tree.Options.DocumentationMode != DocumentationMode.None)
```

Comment:
Is it possible that different trees in the compilation have different `DocumentationMode`s?

Reply:
I don't that is practically possible (though might be theoretically possible to create such a compilation via APIs).